base_report_to_printer/models/printing_printer.py

```diff
@@ -122,35 +75,47 @@ def print_document(self, report, content, format, copies=1):
 
         """
         self.ensure_one()
-        fd, file_name = mkstemp()
-        try:
-            os.write(fd, content)
-        finally:
-            os.close(fd)
-
-        try:
-            _logger.debug(
-                'Starting to connect to CUPS on %s:%s'
-                % (CUPS_HOST, CUPS_PORT))
-            connection = cups.Connection(CUPS_HOST, CUPS_PORT)
-            _logger.debug('Connection to CUPS successfull')
-        except:
-            raise UserError(
-                _("Failed to connect to the CUPS server on %s:%s. "
-                    "Check that the CUPS server is running and that "
-                    "you can reach it from the Odoo server.")
-                % (CUPS_HOST, CUPS_PORT))
-
-        options = self.print_options(report, format, copies)
+
+        connection = self.server_id._openConnection(raise_on_error=True)
+        options = self.print_options(
+            report=report, format=format, copies=copies)
+        # TODO : Check if this works in all cases
+        mime_type = 'application/octet-stream'
```

Comment:
Last thing to check carefully : I didn't found any way to get the mime type of a variable's contents in python.

Does this type work for all file types and all printer drivers ?
If it doesn't, does someone know a way to find the mime type ?